comp/collector/component.go

```diff
@@ -0,0 +1,37 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024-present Datadog, Inc.
+
+// Package collector defines the collector component.
+package collector
+
+import (
+	"github.com/DataDog/datadog-agent/pkg/collector"
+	"github.com/DataDog/datadog-agent/pkg/collector/check"
+	checkid "github.com/DataDog/datadog-agent/pkg/collector/check/id"
+)
+
+// team: agent-metrics-logs
+
+// Component is the component type.
+type Component interface {
+	// Start begins the collector's operation.  The scheduler will not run any checks until this has been called.
+	Start()
```

Comment:
Can you add a TODO to remove `Start` and `Stop` as there is a on going task to remove these kind of methods for components?